src/rust/ensogl/lib/components/src/selector/common/shape.rs

```diff
@@ -0,0 +1,177 @@
+
+        let shape = (rect_left+rect_right+rect_center).into();
+
+        Background{width,height,corner_radius,shape}
+    }
+}
+
+/// Background shape. Appears as a rect with rounded corners. The roundness of each corner can be
+/// toggled by passing `0.0` or `1.0` to either `corner_left` and `corner_right` where `0.0` means
+/// "not rounded" and `1.0` means "rounded".
+pub mod background {
+    use super::*;
+
+    ensogl_core::define_shape_system! {
+        (style:Style,corner_left:f32,corner_right:f32) {
+            let background = Background::new(&corner_left,&corner_right,style);
+            let color      = style.get_color(theme::component::slider::background);
+            let shadow     = shadow::from_shape(background.shape.clone(),style);
+            let background = background.shape.fill(color);
+            (shadow + background).into()
+        }
+    }
+}
+
+
+
+// ===============
+// === IO Rect ===
+// ===============
+
+/// Utility shape that is invisible but provides mouse input. Fills the whole sprite.
+pub mod io_rect {
+    use super::*;
+
+    ensogl_core::define_shape_system! {
+        () {
+            let sprite_width  : Var<Pixels> = "input_size.x".into();
+            let sprite_height : Var<Pixels> = "input_size.y".into();
+
+            let rect  = Rect((&sprite_width,&sprite_height));
+            let shape = rect.fill(HOVER_COLOR);
+
+            shape.into()
+        }
+    }
+}
+
+
+
+// =============
+// === Track ===
+// =============
+
+/// Track of the selector. Appears as filled area of the background. Has a definable start and
+/// end-point (`left`, `right`) which are passed as normalised values relative to the maximum
+/// width.  For consistency with the background shape, also has the property to round either side
+/// of the track, when required to fit the background shape. (See `Background` above.).
+pub mod track {
+    use super::*;
+
+    ensogl_core::define_shape_system! {
+        (style:Style,left:f32,right:f32,corner_left:f32,corner_right:f32,track_color:Vector4) {
+            let background = Background::new(&corner_left,&corner_right,style);
+            let width      = background.width;
+            let height     = background.height;
+
+            let track_width = (&right - &left) * &width;
+            let track_start = left * &width;
+            let track       = Rect((&track_width,&height));
+            let track       = track.translate_x(&track_start + (track_width / 2.0) );
+            let track       = track.translate_x(-width/2.0);
+            let track       = track.intersection(&background.shape);
+
+            let track_color = Var::<color::Rgba>::from(track_color);
+            let track       = track.fill(track_color);
+            track.into()
+          }
+    }
+}
+
+
+
+// ================
+// === OverFlow ===
+// ================
+
+/// Utility struct that contains the overflow shape, and some metadata that can be used to place and
```

Comment:
These docs are not explanatory at all. I want to understand what "overflow shape is" and the docs tell me that "overflow shape is an utility struct that contains the overflow shape" - without any explanation of what it is.